src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1824,19 +1996,25 @@ bool isRefEscape
                     refKind = thisParameter.RefKind;
                 }
 
-                return (thisParameter, receiver, refKind);
+                return new EscapeArgument(thisParameter, receiver, refKind);
             }
 
             static void getArgList(
                 ImmutableArray<BoundExpression> argsOpt,
                 ImmutableArray<RefKind> argRefKindsOpt,
-                ArrayBuilder<(ParameterSymbol? Parameter, BoundExpression Argument, RefKind RefKind)> argsAndParams)
+                ArrayBuilder<MixableDestination>? mixableArguments,
+                ArrayBuilder<EscapeArgument> escapeArguments)
             {
                 for (int argIndex = 0; argIndex < argsOpt.Length; argIndex++)
                 {
                     var argument = argsOpt[argIndex];
                     var refKind = argRefKindsOpt.IsDefault ? RefKind.None : argRefKindsOpt[argIndex];
-                    argsAndParams.Add((null, argument, refKind));
+                    escapeArguments.Add(new EscapeArgument(parameter: null, argument, refKind, isArgList: true));
+
+                    if (refKind == RefKind.Ref && mixableArguments is not null)
```

Comment:
>refKind [](http://example.com/codeflow?start=24&length=7)

Do we need to check if `argument` is a `ref struct`?

Reply:
For better or worse this layer returns all arguments no matter if they are `ref struct` or not. Higher layers filter out the values that aren't going to contribute to escape analysis. I was loath to make a change this deep just now given our time constraints.